blackboxopt/optimizers/botorch_base.py

```diff
@@ -51,13 +51,14 @@ def _acquisition_function_optimizer_factory(
 ) -> Callable[[AcquisitionFunction], Tuple[torch.Tensor, torch.Tensor]]:
     """Prepare either BoTorch's `optimize_acqf_discrete` or `optimize_acqf` depending
     on whether the search space is fully discrete or not and set required defaults if
-    not overridden by `af_opt_kwargs`.
+    not overridden by `af_opt_kwargs`. If any of the af optimizer specific required
+    kwargs are set, this overrides the automatic discrete space detection.
 
     Args:
         search_space: Search space used for optimization.
         af_opt_kwargs: Acquisition function optimizer configuration, e.g. containing
-            values for `n_samples` for discrete optimization, and `num_restarts`,
-            `raw_samples` for the continuous optimization case.
+            values for `num_choices_discrete` for discrete optimization, and
```

Comment:
I find the name `num_choices_discrete` slightly confusing. It could be mistaken for the number of discretisation values in, e.g. grid search. More appropriate names could be `num_samples_discrete`, `num_discrete_samples`, `raw_samples_discrete` or similar.

Reply:
I feel you and started with samples but wanted to be close to the argument names in the botorch function, which calls it choices. What do you think?

Reply:
I can see why it's called choices in BOTorch, because it is already the list of configs from which to choose. We can go with `num_choices_discrete`, it just sounded odd, but if it's close to BOTorch naming, I guess it makes sense.

Reply:
We could also do `num_random_choices` to get the discrete out if that's the problematic one, while still keeping the link to the botorch choices argument.

Reply:
I like `num_random_choices` the best, so I vote for that :)